### PR TITLE
Add track event hook and component

### PR DESCRIPTION
**Pull request by jordanwade**
This PR adds `TrackEventWhenLoaded` component and Amplitude setup. I'm not confident in if this works since I don't have access to Amplitude.

### CODE REVIEW

**Comment by rkdavidson** [comment]
Hey @asleepingpanda and @dzwood , mind giving a second set of eyes on this PR for @jordanwade?

File: pages/_app.js [hunk 1]
```diff
@@ -51,6 +55,27 @@ function App({ Component, pageProps }) {
...
+
+    // NEXT_PUBLIC_AMPLITUDE_KEY  needs to be set in the .env
+    if (!process.env.NEXT_PUBLIC_AMPLITUDE_KEY) {
+      console.warn(
+        'Amplitude Analytics tracking code is required to initialize Amplitude Analytics'
+      );
+      return null;
+    }
+
+    amplitude.init(currentUser);
```
Comment (rkdavidson) [inline pages/_app.js:76]: What about unauthenticated users?
Do you think we need to do any special handling, or pass `null` or something to allow a unique one to be generated for this session? 🤔

**Review by rkdavidson — COMMENTED** [review]
Haven't tested either, but code looks reasonable aside from a question/thought that may be irrelevant.

File: pages/_app.js (hunk 1 above)
Comment (jordanwade) [reply pages/_app.js:76]: Yeah, definitely need to take that in to consideration 👍

**Comment by jordanwade** [comment]
@rkdavidson I was still having issues with the next/dynamic import working. I think it is because it is expecting to be used as a Component...so, I changed it to only import if `window` is defined and seems to work now. 

Feel free to share any knowledge you have on what I might be doing wrong with a new review when you have time. Thank ya!

**Review by rkdavidson — APPROVED** [review]
Tested again and don't see any errors as I trigger misc. events, so I think we're all good!

Feels like follow up work can be tackled separately.

**Pull request merged**